build/base.mk

```diff
@@ -1,7 +1,9 @@
 UNAME_OS := $(shell uname -s)
 UNAME_ARCH := $(shell uname -m)
 
-CACHE_BASE := $(HOME)/.yarpc-go
+XDG_CACHE_HOME ?= $(HOME)/.cache
+
+CACHE_BASE := $(XDG_CACHE_HOME)/yarpc-go
```

Comment:
Make this `?=` so we can set it on Travis. I also don't like making `CACHE_BASE` depend on `XDG_CACHE_HOME` all the time, as this is linux specific. I know you're setting it if it's not set, but the semantics just don't make me feel great - yarpc building isn't linux specific, so if I want to override `CACHE_BASE`, I don't want to have to do it by setting `XDG_CACHE_HOME`.